gettsim/aggregation_numpy.py

```diff
@@ -0,0 +1,147 @@
+import numpy as np
+import numpy_groupies as npg
+
+
+def grouped_count(group_id):
+    fail_if_dtype_of_group_id_not_int(group_id, agg_func="count")
+    out_on_hh = npg.aggregate(
+        group_id, np.ones(len(group_id)), func="sum", fill_value=0
+    )
+
+    out = out_on_hh[group_id]
+    return out
+
+
+def grouped_sum(column, group_id):
+    fail_if_dtype_of_group_id_not_int(group_id, agg_func="sum")
+    fail_if_dtype_not_numeric_or_boolean(column, agg_func="sum")
+
+    out_on_hh = npg.aggregate(group_id, column, func="sum", fill_value=0)
+
+    # Expand to indididual level
+    out = out_on_hh[group_id]
+    return out
+
+
+def grouped_mean(column, group_id):
+    fail_if_dtype_of_group_id_not_int(group_id, agg_func="mean")
+    fail_if_dtype_not_float(column, agg_func="mean")
+
+    out_on_hh = npg.aggregate(group_id, column, func="mean", fill_value=0)
+
+    # Expand to indididual level
+    out = out_on_hh[group_id]
+    return out
+
+
+def grouped_max(column, group_id):
+    fail_if_dtype_of_group_id_not_int(group_id, agg_func="max")
+    fail_if_dtype_not_numeric_or_datetime(column, agg_func="max")
+    fill_value = (
```

Comment:
I added the following:
```
    # numpy_groupies expects a `fill_value` which is assigned to groups which are not
    # present in the data (e.g. if the data only contains the group_id 0 and 4, group
    # 1-3 would get this value). This isn't relevant here because grouped_max returns
    # the values on the individual level. Nevertheless, we need to make sure that the
    # fill_value is convertible to the dtype of `column`.